test/compile.jl

```diff
@@ -71,6 +72,16 @@ try
     end
     Base.compilecache("FooBar")
     sleep(2)
+    @test isfile(joinpath(dir, "FooBar.ji"))
+
+    touch(FooBar_file)
+    insert!(Base.LOAD_CACHE_PATH, 1, dir2)
+    Base.recompile_stale(:FooBar, joinpath(dir, "FooBar.ji"))
+    sleep(2)
```

Comment:
`recompile_stale` is synchronous, so the length of time to do the codegen is not relevant.   `sleep(2)` is only there for paranoia about reading a file on the filesystem that was created by another process which has just completed.